wgpu-hal/src/dx12/conv.rs

```diff
@@ -242,12 +242,12 @@ fn map_blend_factor(factor: wgt::BlendFactor, is_alpha: bool) -> d3d12_ty::D3D12
         Bf::Constant => d3d12_ty::D3D12_BLEND_BLEND_FACTOR,
         Bf::OneMinusConstant => d3d12_ty::D3D12_BLEND_INV_BLEND_FACTOR,
         Bf::SrcAlphaSaturated => d3d12_ty::D3D12_BLEND_SRC_ALPHA_SAT,
-        //Bf::Src1Color if is_alpha => d3d12_ty::D3D12_BLEND_SRC1_ALPHA,
-        //Bf::Src1Color => d3d12_ty::D3D12_BLEND_SRC1_COLOR,
-        //Bf::OneMinusSrc1Color if is_alpha => d3d12_ty::D3D12_BLEND_INV_SRC1_ALPHA,
-        //Bf::OneMinusSrc1Color => d3d12_ty::D3D12_BLEND_INV_SRC1_COLOR,
-        //Bf::Src1Alpha => d3d12_ty::D3D12_BLEND_SRC1_ALPHA,
-        //Bf::OneMinusSrc1Alpha => d3d12_ty::D3D12_BLEND_INV_SRC1_ALPHA,
+        Bf::Src1Color if is_alpha => d3d12_ty::D3D12_BLEND_SRC1_ALPHA,
+        Bf::Src1Color => d3d12_ty::D3D12_BLEND_SRC1_COLOR,
+        Bf::OneMinusSrc1Color if is_alpha => d3d12_ty::D3D12_BLEND_INV_SRC1_ALPHA,
+        Bf::OneMinusSrc1Color => d3d12_ty::D3D12_BLEND_INV_SRC1_COLOR,
```

Comment:
```suggestion
        Bf::Src1 if is_alpha => d3d12_ty::D3D12_BLEND_SRC1_ALPHA,
        Bf::Src1 => d3d12_ty::D3D12_BLEND_SRC1_COLOR,
        Bf::OneMinusSrc1 if is_alpha => d3d12_ty::D3D12_BLEND_INV_SRC1_ALPHA,
        Bf::OneMinusSrc1 => d3d12_ty::D3D12_BLEND_INV_SRC1_COLOR,
```
These seem to have an extra `Color` suffix.

Reply:
On another note, @cwfitzgerald do you know why the DX12 backend is choosing different blend factors based on `if is_alpha` (which is set if we are setting blend factors for the alpha channel)? I don't see any reason for that + none of the other backends do this.

Reply:
Yeah I suspect this is wrong and we always need the alpha blending. Someone needs to cross reference the webgpu spec with https://learn.microsoft.com/en-us/windows/win32/api/d3d12/ne-d3d12-d3d12_blend

Reply:
It seems that for alpha blending: "Blend options that end in _COLOR are not allowed." ([ref](https://learn.microsoft.com/en-us/windows/win32/api/d3d12/ns-d3d12-d3d12_render_target_blend_desc)), so we are doing the right thing here. It doesn't actually matter that this restriction exists since the alpha component is the same between the blend factors that end with `_COLOR` and the ones that end in `_ALPHA`. Vulkan and Metal don't have this restriction since it functionally doesn't matter.

Reply:
What I'm worried about is the other way around - the `COLOR` ones are actually using the color as the blend factor, instead of the alpha - but the documentation isn't terribly clear on what' going on.

Reply:
I'm not sure I understand, could you elaborate?